pkg/bundle/source.go

```diff
@@ -171,10 +180,10 @@ func (b *bundle) secretBundle(ctx context.Context, ref *trustapi.SourceObjectKey
 		if selectorErr != nil {
 			return "", fmt.Errorf("failed to parse label selector as Selector for Secret in namespace %s: %w", b.Namespace, selectorErr)
 		}
-		if err := b.client.List(ctx, &sl, client.MatchingLabelsSelector{Selector: selector}); apierrors.IsNotFound(err) {
```

Comment:
A list call will only return 404 if something is really wrong, ref. https://github.com/kubernetes/client-go/issues/537#issuecomment-450030586

Reply:
Awesome context, thank you!